packages/apollo-link-batch/src/batching.ts

```diff
@@ -102,23 +115,40 @@ export class OperationBatcher {
 
     batchedObservable.subscribe({
       next: results => {
+        if (!Array.isArray(results)) {
+          results = [results];
+          if (nexts.length != 1)
+            console.warn(
+              `server returned single result, expected array of length ${
```

Comment:
Hmm should we have a general thing that makes sure the number of nexts equals the number of results? Looking below, it looks like we don't do anything useful in that case right now.

Reply:
Ah duh, thank you!

I think the reasonable thing to do is propagate an error with the `response` field of the Error set to `results`.